keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionChecker.kt

```diff
@@ -17,21 +18,27 @@ class EnvironmentPromotionChecker(
       .artifacts
       .associateWith { artifactRepository.versions(it) }
       .forEach { (artifact, versions) ->
-        deliveryConfig.environments.forEach { environment ->
-          val version = if (environment.constraints.isEmpty()) {
-            versions.first()
-          } else {
-            versions.first { v ->
-              constraints.all { constraintEvaluator ->
-                !environment.hasSupportedConstraint(constraintEvaluator) || constraintEvaluator.canPromote(artifact, v, deliveryConfig, environment.name)
+        if (versions.isEmpty()) {
```

Comment:
Allow for there being no artifact versions when trying to promote an environment. This didn't break anything but did cause log noise.